### ISSUE TITLE
Mobile app integration test CI

### ISSUE THREAD

**Issue by vaind** [objectbox/objectbox-dart#81]
We should have a way to test ObjectBox runs on mobile (simulator) Android and iOS. From the quick look in the past I think our current tests (/test folder) can't be just executed on the target platform and we would rather need to have an app that instead runs the test suite itself. I've done this quickly in the past with the `/lib/integration_test.dart`. If this is the right approach, it should be moved to a separate dart/flutter project and run from CI for both mobile platforms (when iOS support #47 is merged)

**Comment by vaind** [comment]
Ideally, we should run the complete test suite (`test` folder) on all supported devices, if possible.

**Comment by vaind** [comment]
I've partially prepared this: there's a `flutter-integration-test` job running in GH Actions. Right now, it builds a release apps of both demo applications (sync & non-sync) on mac, windows and Linux.

TODOs:
* run `flutter drive`: Flutter 2.0 has come out with a non-null-safe version of flutter_driver, basically turning it unusable, despite being built-in the SDK :roll_eyes:. As soon as a null-safe version of flutter_driver comes out (hopefully with Flutter 2.1?) we can have a look at the CI and see if it's an option to launch emulator/simulator. If so, `flutter drive` should just work and we can reenable it in `./objectbox/tool/integration-test.sh`
* still untouched - try running the whole test-suit in an app on device. I suspect this could be run with just `lib/main.dart` - looks like a Flutter app doesn't actually have to have any flutter code to be able to run? Let's see...

**Comment by roiskhoiron** [comment]
Is this task already closed? @vaind

**Comment by greenrobot-team** [comment]
@roiskhoiron No, this is still open. Also note that vaind no longer works on this project, so avoid tagging!

**Comment by RaghvindYadav** [comment]
@greenrobot-team Hey I can help your team to resolve this issue. I worked a lot in flutter integrating, widget and unit tests. I also did widget-book implementation.

**Comment by greenrobot-team** [comment]
@RaghvindYadav Thanks for offering! We would welcome a contribution (make sure . As described in the requirements above, do you know if and how the existing unit tests could be run? Is it even possible to test on Android emulators and iOS simulators on GitHub Actions? Would a small proof of concept be a good starting point?

Edit: and maybe have a look at [the contributing info](https://github.com/objectbox/objectbox-dart/blob/main/CONTRIBUTING.md) and make sure you agree that your code would use [the license of this project](https://github.com/objectbox/objectbox-dart/blob/main/LICENSE).

**Comment by thegarvitrai** [comment]
Hey, is this issue still open? I would like to help & assist in this

**Comment by Priyansh956** [comment]
Hi! I'd like to pick this up if it's still open.

A quick update on the flutter_driver blocker mentioned above: it's no longer relevant, and Flutter's integration_test package has since become the standard way to do this (it's null-safe, ships with the SDK, and runs via a normal flutter test target instead of flutter drive). It's designed for exactly this use case, including apps with little/no UI.

My proposed plan:

1. Let's create a separate minimal Flutter app project (as suggested in the issue) with a bare-bones lib/main.dart and an integration_test/ test file that runs the existing objectbox_test suite (or the relevant subset) on-device.

2. Wire it into the existing flutter-integration-test CI job:
- Android: boot an emulator in CI and run flutter test integration_test.
- iOS: hold off until #47 lands, then do the same with a booted simulator.

Update objectbox/tool/integration-test.sh to call flutter test integration_test instead of the disabled flutter drive step.

I'll start with Android only, then follow up with iOS once that's in place. Happy to adjust the approach if you have a different idea in mind — let me know if I can take this on!

**Comment by greenrobot-team** [comment]
@Priyansh956 Thanks. However, as part of a new feature we have already created a new integration test (Android Flutter) app [using the new integration_test package](https://docs.flutter.dev/testing/integration-tests). It should make it into a future public release.
